crates/shared/src/event_handling.rs

```diff
@@ -282,9 +288,12 @@ where
         if let Some(range) = event_range.history_range {
             self.update_events_from_old_blocks(range).await?;
         }
-        if !event_range.latest_blocks.is_empty() {
+        if let Some(last_block) = event_range.latest_blocks.last() {
             self.update_events_from_latest_blocks(&event_range.latest_blocks, event_range.is_reorg)
                 .await?;
+            self.store_mut()
+                .persist_last_processed_block(last_block.0)
```

Comment:
nit: I'd move this call inside `update_last_handled_blocks` as this function is where we update the local cache so aligning local cache and db makes sense.